Rework iptables mod dependencies, rename to ip(6)tables-legacy, add PROVIDES

champtar opened the pull request:
This PR clean up a bit more iptables dependencies.
This is a breaking changes, packages selecting iptables-mod-* will not pull iptables anymore, so they might want to add `+iptables` as dependency.

Packages should not depend on iptables-legacy or iptables-nft directly to allow people to switch between them easily (and between fw3/fw4).
We will still need to rework netfilter part to not always pull the iptable* kmods (legacy tables) but this doesn't prevent the system to properly work.

To have nft properly display rules created by iptables-nft would need to compile nft with `--with-xtables` and remove `IPT_BUILTIN` hack but it's not needed for properly functioning system and will likely increase disk usage, so not sure it's worth the effort.

champtar committed:
iptables: make mod depend on libxtables

'iptables-mod-' can be used directly by firewall3, by
iptables and by iptables-nft. They are not linked to
iptables but to libxtables, so fix the dependencies to allow
to remove iptables(-legacy)

Signed-off-by: Etienne Champetier <champetier.etienne@gmail.com>

champtar committed:
iptables: move IPTABLES_{CONNLABEL,NFTABLES} to libxtables

Those 2 configs are not specific to iptables(-legacy)

Signed-off-by: Etienne Champetier <champetier.etienne@gmail.com>

champtar committed:
iptables: rename to ip(6)tables-legacy, add PROVIDES

Using PROVIDES allows to have other packages continue to
depend on iptables and users to pick between legacy and nft
version.

Signed-off-by: Etienne Champetier <champetier.etienne@gmail.com>

hnyman commented on package/network/utils/iptables/Makefile:
```diff
@@ -113,6 +99,7 @@ define Package/iptables-nft
 $(call Package/iptables/Default)
   TITLE:=IP firewall administration tool nft
   DEPENDS:=@IPTABLES_NFTABLES +libxtables-nft +libip4tc +IPV6:libip6tc +kmod-ipt-core +kmod-nft-compat
```
I can't test for the next few days, but the changes look ok to me, at the first glance.

One thing that you might investigate, is the IPTABLES_NFTABLES dependency for iptables-nft. Is that symbol actually needed for libxtables?

champtar replied:
I moved `libnftnl` dependency to `iptables-nft`, and added `@IPTABLES_NFTABLES` to `libiptext-nft (libxtables-nft)`

champtar committed:
iptables: fix libnftnl/IPTABLES_NFTABLES dependency

libxtables doesn't depend on libnftnl, iptables-nft does,
so move the dependency to not pull libnftnl with firewall3/iptables-legacy

Also libxtables-nft depends on IPTABLES_NFTABLES

Signed-off-by: Etienne Champetier <champetier.etienne@gmail.com>

champtar committed:
iptables: move libiptext* to their own packages

iptables-nft doesn't depend on libip{4,6}tc, so move
libiptext* libs in their own packages to clean up dependencies
Rename libxtables-nft to libiptext-nft

Signed-off-by: Etienne Champetier <champetier.etienne@gmail.com>

champtar committed:
iptables: bump PKG_RELEASE

Following dependencies rework, bump PKG_RELEASE

Signed-off-by: Etienne Champetier <champetier.etienne@gmail.com>

hauke commented:
Thank you for the patches, I applied them to master in 50d327196681f9514739355573e49db0ad151d6a, d35a573004ff21e5a7bbdb896967454c63b7a363, 316c406e62a535d17d654ab8c4f5e84a0485058f, 795e7155cbe3e78669f6821bf7aecb7c4e1e1afb, 39d50a2008b7f29a46b7a9511525613816cc15e1 and d95b74f7c9f8ce58e583991f598a9b39bb43ee2f
